src/jwk.rs

```diff
@@ -0,0 +1,79 @@
+use askar_crypto::jwk::JwkParts;
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+
+use crate::error::{ErrorKind, Result, ResultExt};
+
+/// TODO: FIXME: Think on how to integrate better with JwkParts<'a>
+/// without re-serialization
+#[derive(Debug, Clone)]
+pub(crate) struct JWK(String);
+
+impl JWK {
+    pub fn _parts(&self) -> Result<JwkParts> {
+        JwkParts::from_str(&self.0).kind(ErrorKind::Malformed, "jwl malformed.")
+    }
+}
+
+impl Serialize for JWK {
+    fn serialize<S>(&self, s: S) -> std::result::Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        let value: Value =
+            serde_json::from_str(&self.0).map_err(|e| serde::ser::Error::custom(e.to_string()))?;
+
+        Serialize::serialize(&value, s)
+    }
+}
+
+impl<'de> Deserialize<'de> for JWK {
+    fn deserialize<D>(d: D) -> std::result::Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        let jwk: Value = Deserialize::deserialize(d)?;
+
+        let jwk =
+            serde_json::to_string(&jwk).map_err(|e| serde::de::Error::custom(e.to_string()))?;
+
+        Ok(JWK(jwk))
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
```

Comment:
It would be great to add some negative tests with invalid JWKs

Reply:
I plan to remove this helper in the next update. aries-crypto merged improvements in JWK serialisation.

Reply:
Tests added